gtk4/src/subclass/buildable.rs

```diff
@@ -51,35 +51,36 @@ pub trait BuildableImpl: ObjectImpl {
     fn construct_child(&self, buildable: &Self::Type, builder: &Builder, name: &str) -> Object {
         self.parent_construct_child(buildable, builder, name)
     }
-    /*
-    Only useful for custom tags, not something the application developer has to often implement
-    and needs more thinking in terms of how to handle both BuildableParser & the various ptr you're supposed to pass around
-    fn custom_tag_start(
+    unsafe fn custom_tag_start(
         &self,
         buildable: &Self::Type,
         builder: &Builder,
         child: Option<&Object>,
-        tagname: &str,
-        parser: BuildableParser,
-        data: ptr,
-    );
-    fn custom_tag_end(
+        tag_name: &str,
+        parser_data: *mut *mut libc::c_void,
```

Comment:
How is this pointer used exactly?

Reply:
It is used to pass the state to the BuildableParser, my idea was to create a struct that implements a certain trait and move the struct around instead of passing a random ptr around

Reply:
Sounds like a good plan :) Where would the struct be freed, is the `finish()` function guaranteed to be called exactly once with the same parser data, for example?